src/components/shopping-cart/index.js

```diff
@@ -56,48 +78,94 @@ export default class ShoppingCart extends React.PureComponent {
   }
 
   renderItemList = () => {
+    const { order } = this.props
     const itemIds = this.getItemIds()
+
+    if (!this.state.isExpanded || !order) {
+      return null
+    }
+
     return (
       <Flex column css={style.itemsList}>
         <p css={style.itemsListTitleText}>Items</p>
         {itemIds.map(itemId => (
-          <ItemRow {...{ itemId }} key={itemId} />
+          <ItemRow
+            {...{
+              itemId,
+              isEditable: orderUtil.isWaitingForTransaction(order),
+            }}
+            key={itemId}
+          />
         ))}
       </Flex>
     )
   }
 
-  renderQRCode = () => (
-    <Flex css={style.qrCode__maxHeightWrapper}>
-      <Flex css={style.qrCode__sqaureWrapper}>
-        <Flex css={style.qrCode__innerFillWrapper}>
-          <Flex css={style.qrCode}>
-            <QRCode {...this.props} />
+  renderPayment = () => {
+    const { order } = this.props
+
+    if (!this.state.isExpanded || !order) {
+      return null
+    }
+
+    return (
+      <Flex css={style.qrCode__maxHeightWrapper}>
+        <Flex css={style.qrCode__sqaureWrapper}>
+          <Flex css={style.qrCode__innerFillWrapper}>
+            <Flex css={style.qrCode}>
+              {orderUtil.isWaitingForTransaction(order) ? (
+                <QRCode {...this.props} />
+              ) : (
+                <Confirmation
+                  {...{ order }}
+                  onSellMoreItemsClick={this.handleSellMoreItemsClick}
+                />
+              )}
+            </Flex>
           </Flex>
         </Flex>
       </Flex>
-    </Flex>
-  )
+    )
+  }
 
-  // //////////////////
+  /////////////////////
+  // LIFECYCLE HOOKS //
+  /////////////////////
+
+  componentDidUpdate = (prevProps: PropsType) => {
+    if (this.getItemCount(prevProps) !== 0 && this.getItemCount() === 0) {
+      // this.setState({ isExpanded: false })
+    }
+  }
+
+  ////////////////////
   // EVENT HANDLERS //
-  // //////////////////
+  ////////////////////
 
   handleHeaderClick = () => {
     this.setState(prevState => ({
       isExpanded: !prevState.isExpanded,
     }))
   }
 
-  // ///////////
+  handleSellMoreItemsClick = () => {
+    this.setState({
+      isExpanded: false,
+    })
+    this.props.onResetShoppingCart()
+  }
+
+  /////////////
   // GETTERS //
-  // ///////////
+  /////////////
 
-  getItemIds = () => Object.keys(_.get(this.props, 'order.items', {}))
+  getItemIds = (): Array<IdType> =>
```

Comment:
$ReadOnly or nah?

Reply:
I don't think we should worry about it

Reply:
React prints warnings when you try to write in props